WordPressAuthenticator/GoogleSignIn/URL+GoogleSignIn.swift

```diff
@@ -2,26 +2,50 @@ import Foundation
 
 extension URL {
 
-    // TODO: This is incomplete
-    static func googleSignInAuthURL(clientId: String) throws -> URL {
-        let baseURL = "https://accounts.google.com/o/oauth2/v2/auth"
+    // It's acceptable to force-unwrap here because, for this call to fail we'd need a developer
+    // error, which we would catch because the unit tests would crash.
+    static var googleSignInBaseURL = URL(string: "https://accounts.google.com/o/oauth2/v2/auth")!
 
+    static func googleSignInAuthURL(clientId: String, pkce: ProofKeyForCodeExchange) throws -> URL {
```

Comment:
Good point. I was thinking about this before going AFK and realized I may be using unclear boundaries here.

You might have noticed I made an `OAuth` folder and a `GoogleSignIn` one. My intent was to separate the components that are GoogleSignIn specific from those that are part of the OAuth spec. One issue with how I executed this is that I haven't been cross checking between the OAuth and Google documentations so there might be stuff in the OAuth folder that's still Google specific, and vice versa.

Thinking about it since, I came to the conclusion that I'd prefer to separate OAuth- from Google- specific code only once we'll need to add a new OAuth client (likely WordPress OAuth2). As such, I'm tempted to follow up this PR with one that moves all the `OAuth/` code into `GoogleSignIn/`, renaming some of the types if necessary.

My rationale for the suggestion above is that I'm more interested in having a working version of the SDK-less authentication than a general purpose OAuth client, even though building a client first might make it easier to add further authentication implementation in the future.

What do you think?

Reply:
Sounds good to me. We probably don't need to "pre-abstract" stuff if there is no immediate plan to support another OAuth provider (service? server?). I trust you'll strike the right balance 👍

Reply:
> I trust you'll strike the right balance 👍

Thanks for the confidence.